src/components/draw/DrawDirective.js

```diff
@@ -27,889 +28,798 @@ goog.require('ga_permalink');
-            scope.$evalAsync();
-          };
-
-
-
-          ////////////////////////////////////
-          // Tab managment
-          ////////////////////////////////////
-          scope.activeTabProps = function() {
-            scope.isPropsActive = true;
-            scope.isMeasureActive = false;
-            scope.options.isProfileActive = false;
-          };
-          scope.activeTabProfile = function() {
-            scope.isPropsActive = false;
-            scope.isMeasureActive = false;
-            scope.options.isProfileActive = true;
-          };
-          scope.activeTabMeasure = function() {
-            scope.isPropsActive = false;
-            scope.isMeasureActive = true;
-            scope.options.isProfileActive = false;
-          };
-          scope.showTabs = function(feature) {
-            var bools = [
-              scope.showPropsTab(feature),
-              scope.showMeasureTab(feature),
-              scope.showProfileTab(feature)
-            ];
-            var cpt = 0;
-            for (var i = 0; i < bools.length; i++) {
-              if (bools[i]) {
-                 cpt++;
-              }
-            }
-            return (cpt > 1);
-          };
-          scope.showMeasureTab = function(feature) {
-            var geom = feature.getGeometry();
-            var isPoint = (geom instanceof ol.geom.Point);
-            return (!gaMapUtils.isMeasureFeature(feature) && !isPoint);
-          };
-          scope.showProfileTab = function(feature) {
-            return scope.showMeasureTab(feature);
-          };
-          scope.showPropsTab = function(feature) {
-            var geom = feature.getGeometry();
-            var isPoint = (geom instanceof ol.geom.Point);
-            if (isPoint) {
-              scope.activeTabProps();
-            }
-            return (!isPoint && !gaMapUtils.isMeasureFeature(feature));
-          };
+        scope.canExport = function() {
+          return (layer) ? layer.getSource().getFeatures().length > 0 : false;
+        };
 
 
+        ////////////////////////////////////
+        // Show share modal
+        ////////////////////////////////////
+        scope.canShare = function() {
+          return layer && layer.adminId;
+        };
+        scope.share = function(evt) {
+          if (evt.currentTarget.attributes.disabled) {
+            return;
+          }
+          $rootScope.$broadcast('gaShareDrawActive', layer);
+        };
 
 
-          ////////////////////////////////////
-          // create/update the file on s3
-          ////////////////////////////////////
-          var save = function(evt) {
-            if (layer.getSource().getFeatures().length == 0) {
-              //if no features to save do nothing
-              return;
+        ////////////////////////////////////
+        // Popup content management
+        ////////////////////////////////////
+        var togglePopup = function(feature, clickCoord) {
```

Comment:
It's more than toggle functionality now. Maybe the function should be renamed. Minor.